src/Silverback.Integration/Messaging/Encryption/SymmetricEncryptionSettings.cs

```diff
@@ -59,6 +60,16 @@ public class SymmetricEncryptionSettings : EncryptionSettings
         [SuppressMessage("", "CA1819", Justification = Justifications.CanExposeByteArray)]
         public byte[]? Key { get; set; }
 
+        /// <summary>
+        ///     Gets or sets the key identifier.
+        /// </summary>
+        public string? KeyIdentifier { get; set; }
+
+        /// <summary>
+        /// Gets or sets the callback function to retrieve the decryption key.
+        /// </summary>
+        public Func<string, byte[]>? DecryptionKeyCallback { get; set; }
```

Comment:
1. In general, I wouldn't call this `DecryptionKeyProvider` instead of `Callback`. Everywhere.
2. IMHO, we should only allow to set either the fixed key or the callback function. Therefore you should add a check in the `Validate`.

Reply:
I renamed and I added the check in the `Validate` method of `DecryptionWithKeyRotationSettings` class.